src/Core/Services/Implementations/CollectionService.cs

```diff
@@ -53,21 +53,40 @@ public class CollectionService : ICollectionService
         }
 
         var groupsList = groups?.ToList();
-        var usersList = users?.ToList();
+        var usersList = users?.ToList() ?? new List<CollectionAccessSelection>();
 
         // If using Flexible Collections - a collection should always have someone with Can Manage permissions
         if (_featureService.IsEnabled(FeatureFlagKeys.FlexibleCollections, _currentContext))
         {
             var groupHasManageAccess = groupsList?.Any(g => g.Manage) ?? false;
-            var userHasManageAccess = usersList?.Any(u => u.Manage) ?? false;
+            var userHasManageAccess = usersList.Any(u => u.Manage);
             if (!groupHasManageAccess && !userHasManageAccess)
             {
                 throw new BadRequestException(
                     "At least one member or group must have can manage permission.");
             }
         }
+        else
+        {
+            // If not using Flexible Collections
+            // all users with EditAnyCollection permission should have Can Manage permission for the collection
+            var organizationUsers = await _organizationUserRepository
+                .GetManyByOrganizationAsync(collection.OrganizationId, null);
+            foreach (var orgUser in organizationUsers.Where(ou => ou.GetPermissions()?.EditAnyCollection ?? false))
+            {
+                var user = usersList.FirstOrDefault(u => u.Id == orgUser.Id);
+                if (user != null)
+                {
+                    user.Manage = true;
+                }
+                else
+                {
+                    usersList.Add(new CollectionAccessSelection { Id = orgUser.Id, Manage = true });
+                }
+            }
```

Comment:
My bad - I see I mixed up `EditAny` and `EditAssigned` in my breakdown.

We're not deprecating `EditAny`, only `EditAssigned`.

So this logic should only set `user.Manage = true` if they're already being assigned access. In other words:
* check the `EditAssignedCollection` permission, not `Any`
* delete the `usersList.Add` block

Then I think it should work? But just double check for me.

Reply:
I have revisited this, it should now be according to your comment above